### PR TITLE
issue #635: Add django-anymail

### PR DESCRIPTION
**Pull request by vaishnavi-2901**
## Description
I added a depedencies and some base code to use django-anymail

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires dependencies updates.
- [x] I have updated the dependencies accordingly.

#Note: 
Please update the config file for 
1. SENDGRID_API_KEY (this needs to be get via sendgrid)
2. DEFAULT_FROM_EMAIL

### CODE REVIEW

File: backend/project_name/settings/base.py [hunk 1]
```diff
@@ -250,3 +251,16 @@ def base_dir_join(*args):
...
+# myproject/settings.py
+
+# Email Backend Configuration
+EMAIL_BACKEND = "anymail.backends.sendgrid.EmailBackend"  # or another supported backend
+ANYMAIL = {
+    "SENDGRID_API_KEY": config("SENDGRID_API_KEY"),  # Replace with your actual API key
+    # can add other sendgrid fields here
+}
+
+DEFAULT_FROM_EMAIL = config("DEFAULT_FROM_EMAIL")  # Replace with your default from email
```
Comment (fjsj) [inline backend/project_name/settings/base.py:266]: Instead of changing base.py to use sendgrid, please change production.py.

**Review by fjsj — CHANGES_REQUESTED** [review]
Thanks for the PR. Please check comments. Also, please test on production to ensure it's working properly.

**Comment by vishu1089** [comment]
Okay

**Comment by vishu1089** [comment]
For that, I need ENV file

**Comment by fjsj** [comment]
You'll follow the README to produce your own env to test.

**Comment by vishu1089** [comment]
Oh, that I did already
I thought we need to test with real creds.

Will move it to production.py

**Comment by hugobessa** [comment]
This change is probably breaking CI and render.com setup as it introduces new required env vars that aren't mentioned there.

**Comment by vishu1089** [comment]
Yes, mentioned the same on Note under description. Should I ping env file
as well?

**Comment by fjsj** [comment]
@vishu1089 you'll have to update render.yaml to not use SENDGRID_USERNAME and SENDGRID_PASSWORD and start using SENDGRID_API_KEY.

CTRL+F everywhere in the project for sendgrid and update whatever is necessary too, please.

**Comment by vishu1089** [comment]
Will do, thanks for the heads up.

**Comment by vaishnavi-2901** [comment]
done, please recheck

File: README.md [hunk 2]
```diff
@@ -290,7 +290,7 @@ Uncommenting the worker configuration lines on `render.yaml` will imply in costs
 
 ### SendGrid
 
-To enable sending emails from your application you'll need to have a valid SendGrid account and also a valid verified sender identity. After finishing the validation process you'll be able to generate the API credentials and define the `SENDGRID_USERNAME` and `SENDGRID_PASSWORD` environment variables on Render.com.
+To enable sending emails from your application you'll need to have a valid SendGrid account and also a valid verified sender identity. After finishing the validation process you'll be able to generate the API credentials and define the `SENDGRID_API_KEY` and `SENDGRID_PASSWORD` environment variables on Render.com.
```
Comment (fjsj) [inline README.md:293]: Remove mention to SENDGRID_PASSWORD.

File: backend/project_name/settings/production.py [hunk 3]
```diff
@@ -22,9 +22,9 @@
 
 SERVER_EMAIL = "foo@example.com"
 
-EMAIL_HOST = "smtp.sendgrid.net"
-EMAIL_HOST_USER = config("SENDGRID_USERNAME")
-EMAIL_HOST_PASSWORD = config("SENDGRID_PASSWORD")
+# EMAIL_HOST = "smtp.sendgrid.net"
+# EMAIL_HOST_USER = config("SENDGRID_USERNAME")
+# EMAIL_HOST_PASSWORD = config("SENDGRID_PASSWORD")
```
Comment (fjsj) [inline backend/project_name/settings/production.py:27]: Remove commented code.

File: backend/project_name/settings/production.py [hunk 4]
```diff
@@ -135,3 +135,12 @@
 
 # Sentry
 sentry_sdk.init(dsn=SENTRY_DSN, integrations=[DjangoIntegration()], release=COMMIT_SHA)
+
+# Email Backend Configuration
+EMAIL_BACKEND = "anymail.backends.sendgrid.EmailBackend"  # or another supported backend
+ANYMAIL = {
+    "SENDGRID_API_KEY": config("SENDGRID_API_KEY"),  # Replace with your actual API key
```
Comment (fjsj) [inline backend/project_name/settings/production.py:142]: Remove "# Replace with your actual API key"
There's no need to replace. Dev should set the env var.

File: backend/project_name/settings/production.py [hunk 5]
```diff
@@ -135,3 +135,12 @@
...
+    # can add other sendgrid fields here
+}
+
+DEFAULT_FROM_EMAIL = config("DEFAULT_FROM_EMAIL")  # Replace with your default from email
```
Comment (fjsj) [inline backend/project_name/settings/production.py:146]: Remove "# Replace with your default from email"
There's no need to replace. Dev should set the env var.

**Review by fjsj — CHANGES_REQUESTED** [review]
Check comments

File: backend/project_name/settings/base.py (hunk 1 above)
Comment (vaishnavi-2901) [reply backend/project_name/settings/base.py]: done, resolving it now

File: README.md (hunk 2 above)
Comment (vaishnavi-2901) [reply README.md]: done, resolving it now

File: backend/project_name/settings/production.py (hunk 4 above)
Comment (vaishnavi-2901) [reply backend/project_name/settings/production.py]: done, resolving it now

File: backend/project_name/settings/production.py (hunk 5 above)
Comment (vaishnavi-2901) [reply backend/project_name/settings/production.py]: done, resolving it now

**Comment by vaishnavi-2901** [comment]
all the comments are resolved, please re-review

**Review by fjsj — APPROVED** [review]
Overall LGTM, but I will test manually before merging. Thanks.

**Review by fjsj — CHANGES_REQUESTED** [review]
After a closer review, there are still errors on this PR. You didn't update `proj_main.yml` removing SENDGRID_USERNAME and SENDGRID_PASSWORD and adding SENDGRID_API_KEY there. Also update `.github/workflows/shared-build/action.yml`.

While we appreciate the PR, please be extra-careful on changing settings. I had requested a full search over all files for "sendgrid" to perform all the changes. Please ensure you did that.

**Comment by fjsj** [comment]
Additionally, remove the poetry.lock file. We don't include that in the boilerplate, and it's risky to keep it anyway due to supply-chain attacks.

**Comment by vaishnavi-2901** [comment]
Okay

**Comment by fjsj** [comment]
Closing due to inactivity. Please feel free to reopen with the issues addressed.

**Pull request closed without merging**